File: Data Structure/String/Easy/LET_no1544_Make The String Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        res = []
        i = 0
        while i < len(s):
            if res and res[-1] != s[i] and res[-1].lower() == s[i].lower():
                res.pop()
                i += 1
                continue
            
            res.append(s[i])
            i += 1
                
        return ''.join(res)

File: Data Structure/String/Easy/LET_no1544_Make The String Great.cpp
class Solution {
public:
    string makeGood(string s) {
        int n = s.size(), i = 0;
        string res = "";
        while (i < n) {
            if (!res.empty() && res.back() != s[i] && tolower(res.back()) == tolower(s[i])) {
                res.pop_back();
                i++;
                continue;
            }

            res.push_back(s[i++]);
        }

        return res;
    }
};